tests/test_sequence_interpolator.py

```diff
@@ -0,0 +1,429 @@
+
+                expected[i] = y1 + ((t - t1) / (t2 - t1)) * (y2 - y1)
+
+            interp, valid = seq_interp.interpolate(times=shifted_times)
+
+            valid_indices = np.where(valid)[0]
+            if len(valid_indices) > 0:
+                assert np.allclose(
+                    interp[valid_indices, sig_idx], expected[valid_indices], atol=1e-6
+                ), f"Linear interpolation mismatch for signal {sig_idx}"
+
+
+@pytest.mark.parametrize("interpolation_mode", ["nearest_neighbor", "linear"])
+@pytest.mark.parametrize("keep_nans", [False, True])
+def test_interpolation_for_invalid_times(interpolation_mode, keep_nans):
+    end_time = 5.0
+    with sequence_data_and_interpolator(
+        data_kwargs=dict(
+            n_signals=10,
+            use_mem_mapped=True,
+            t_end=end_time,
+            sampling_rate=10.0,
+        ),
+        interp_kwargs=dict(keep_nans=keep_nans),
+    ) as (_, _, _, seq_interp):
+        assert isinstance(
+            seq_interp, SequenceInterpolator
+        ), "Interpolation object is not a SequenceInterpolator"
+        seq_interp.interpolation_mode = interpolation_mode
+
+        times = np.array([-5.0, -0.1, 0.1, 4.9, 5.0, 5.1, 10.0])
+        interp, valid = seq_interp.interpolate(times=times)
+        expected_valid = (
+            np.where((times >= 0.0) & (times <= end_time))[0]
+            if interpolation_mode == "nearest_neighbor"
+            else np.where((times >= 0.0) & (times < end_time))[0]
+        )
+        assert (
+            expected_valid == valid
+        ).all(), "Valid times does not match expected values"
+        assert interp.shape == (
+            3,
+            10,
+        ), f"Expected interp.shape == (2, 10), got {interp.shape}"
+
+
+@pytest.mark.parametrize("interpolation_mode", ["nearest_neighbor", "linear"])
+@pytest.mark.parametrize("end_time", [0.2, 1.0, 5.0])
+@pytest.mark.parametrize("keep_nans", [False, True])
+def test_interpolation_with_phase_shifts_for_invalid_times(
+    interpolation_mode, end_time, keep_nans
+):
+    with sequence_data_and_interpolator(
+        data_kwargs=dict(
+            n_signals=10,
+            use_mem_mapped=True,
+            t_end=end_time,
+            sampling_rate=10.0,
+            shifts_per_signal=True,
+        ),
+        interp_kwargs=dict(keep_nans=keep_nans),
+    ) as (timestamps, _, _, seq_interp):
+        assert isinstance(
+            seq_interp, PhaseShiftedSequenceInterpolator
+        ), "Interpolation object is not a PhaseShiftedSequenceInterpolator"
+        seq_interp.interpolation_mode = interpolation_mode
+
+        times = np.array([-5.0, -0.1, 0.1, 4.9, 4.9999999, 5.0, 5.0000001, 5.1, 10.0])
+        interp, valid = seq_interp.interpolate(times=times)
+        assert (
+            np.where((times >= 0.09) & (times <= end_time + 0.09))[0] == valid
+        ).all(), "Valid times does not match expected values"
```

Comment:
why do we add sth for the `end_time` here?
We do not do it like this in the intervals.intersect ([link](https://github.com/sensorium-competition/experanto/blob/main/experanto/intervals.py#L27-L28))
```
def intersect(self, times: np.ndarray) -> np.ndarray:
        return np.where((times >= self.start) & (times <= self.end))[0]
```
and if 0.09 is start_time - could we give it a name and then use:) (for the sake of clarity)
```suggestion
        start_time = 0.09
        assert (
            np.where((times >= start_time) & (times <= end_time))[0] == valid
        ).all(), "Valid times does not match expected values"
```

Reply:
For the phase shifts. They are not added in `intervals.intersect`, but here: https://github.com/sensorium-competition/experanto/blob/cac5668bf23d8c5969969e9a5ffb795788b593e5/experanto/interpolators.py#L242

I made it clearer.